### DISCUSSION TITLE
How can I dynamically switch between row selection and inline editing (with enableExcelCopyBuffer)?

### DISCUSSION THREAD

**Discussion by Harsgalt86** [ghiscoding/aurelia-slickgrid#1120] [Q&A]
/**
   * Defaults to true, which leads to use an Excel like copy buffer that gets copied in clipboard and can be pasted back in Excel or any other app.
   * NOTE: please note that this option will NOT work when "Row Selection" & "Row Move" are enabled, because features are conflicting with each other.
   */
  enableExcelCopyBuffer?: boolean;

Because of the limitation with `enableExcelCopyBuffer` we're looking at having a toggle that switches between an edit and a select mode. When `isEditMode` changes we reset the `gridOptions` and call slickGrid.init(). This only works but only when going from select mode to edit. When going the other way, the checkboxes do not reappear?

**Comment by Harsgalt86** [comment]
I'd like to avoid doing something like
```
<aurelia-slickgrid 
 if.bind="editModeOn"
  ...
>
<aurelia-slickgrid 
 if.bind="!editModeOn"
  ...
>
```

**Comment by ghiscoding** [comment]
The reason is because they both use different Selection Models (`SlickRowSelectionModel` vs `SlickCellSelectionModel`), you could maybe try to instantiate new ones when switching mode. Someone posted a similar approach on Stack Overflow, I never tried it and I'm not sure that really works without confusing the Dependency Injection, you could try it and don't call `grid.init()` as much as possible because that's probably a bad idea since it resets a few things

https://stackoverflow.com/a/76810071/1212166

Out of topic, I left some comments in your opened PR and waited to do a release but I want to do one in the coming days with or without your PR. I'm getting closer to a Slickgrid-Universal new major alpha release and want current master branch to be fully released before a major alpha release, however on the Aurelia side the next major (before Christmas) will migrate and require Aurelia 2. I also don't usually support older version once new major versions are out, I just got too many libs to support for free

**Reply by Harsgalt86** [reply]
That's all good, I'm going to have to put that PR aside for the time being. I might revisit if there's time for bug fixing, but unfortunately work priorities have changed.
